code-challenges.test.js

```diff
@@ -35,35 +51,410 @@ const num2 = 0
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango", "deduction"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango", "Deduction"]
 
-
 // b) Create the function that makes the test pass.
 
+// create a function called capitalWords
+// with an array as parameter
+// declare capWords array as empty
+// use for loops iteration to go into each word of the array
+//use charAt(0) and .toUppercase to capitalize the first letter of
+//of each value of each string array and add the rest of the letters ///of the string with the .slice
+// console.log to input the capitalized word in the capWord array
+// after the iteration, use return capWord to output the capWord 
+//array with capitalized first letter
 
 
-// --------------------3) Create a function that takes in a string and logs the index of the first vowel.
+const capitalWords = (array) => {
+    let capWords = []
+    for (var i = 0; i < array.length; i++ {
+     array[i] = array[i].charAt(0).toUpperCase() + array[i].slice(1)
+     console.log(capWords)
+}
+    return capWords
+console.log(capitalWords(randomNouns1))
+console.log(capitalWords(randomNouns2))
+console.log(capitalWords(randomNouns3))
+}
 
+//Test failed and got stuck again here.
+
+// --------------------3) Create a function that takes in a string and logs the index of the first vowel.
+// Pseudo code: create a function called firstVowel 
+// input ===> takes in a string
+// output ===> the index of the first vowel
 // a) Create a test with expect statements for each of the variables provided.
 
+describe ("firstVowel", () => {
+    it("returns the index of the first vowel of the string", () => {
+        expect(vowelTester1("learn")).toEqual("1")
+        expect(vowelTester2("academy")).toEqual("0")
+        expect(vowelTester3("challenges")).toEqual("2")
+
+    })
+})
+
 const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
-
+//First test failed.
+//create function using
 
 // b) Create the function that makes the test pass.
+// Pseudo code:
+// create a function called firstVowel
+// the function takes on a parameter composed of string
+// use the search function to search for the first vowel
+// used the return[i] to output the index of the vowel
+// used console.log to invoke the function to get the index of the /////first vowel
+
+function firstVowel(str)  {
+    let test = vowelTest.search(/[a,e,i,o,u,A,E,I,O,U]/g)
+    return [i]
+}
+    console.log(firstVowel(vowelTester1()))
+    console.log(firstVowel(vowelTester2()))
+    console.log(firstVowel(vowelTester3()))
+
+
+//Yarn jest test failed again. Tried to very for spelling, misplaced punctuations or characters after the 3 failed tests. Tried to compare code with some of those discussed in class but still can not figure out the error.
```

Comment:
Have to be mindful of datatype. What datatype then you tell your test to expect? What datatype does your function output?